sys/newlib/syscalls.c

```diff
@@ -203,11 +203,13 @@ int _open_r(struct _reent *r, const char *name, int mode)
 int _read_r(struct _reent *r, int fd, void *buffer, unsigned int count)
 {
 #ifndef MODULE_UART0
+    int res;
     mutex_lock(&uart_rx_mutex);
-
+    unsigned state = disableIRQ();
     count = count < rx_buf.avail ? count : rx_buf.avail;
-
-    return ringbuffer_get(&rx_buf, (char*)buffer, count);
+    res = ringbuffer_get(&rx_buf, (char*)buffer, count);
```

Comment:
hm, before we just returned ringbuffer_get directly. And the compiler doesn't complain. And no RIOT device in the near future will have a ring buffer with 2^31 bytes, so not casting should be fine. :)